Purpose: Internet Archive already allows for the downloading of a csv through: https://archive.org/advancedsearch.php. However, if there are multiple entries for a metadata element, the column will be formated as a list (['heading 1', 'heading 2']). This script removes the list format so the column is in the following format by default ('heading 1'; 'heading 2'). It also allows the user to set the delimiter. Finally, Internet Archive's download only allows for a keyword search. This script allows for any advanced search of the title, creator and date.

1\. Run script

2\. Choose search type (K or A), Hit Enter

3\. For Keyword: Enter Keyword search, hit enter

4\. Enter Delimiter, leave blank for ; 

```
Examples: 
Creator with entered delimiter
Degas, Edgar, 1834-1917* Toulouse-Lautrec, Henri de, 1864-1901* Roberts, Keith, 1937-
Creator with delimiter left blank
Degas, Edgar, 1834-1917; Toulouse-Lautrec, Henri de, 1864-1901; Roberts, Keith, 1937-
```

5\. Hit Enter.

A csv file named with the search query will be in the same folder as the script. 

In [6]:
python ia_search.py

Keyword or Advanced (Title, Creator, Mediatype) Search (K/A): K
Enter Query: Edgar Degas
Enter Delimiter, leave blank for semi-colon: 
150 records retrieved


3\. For Advanced Search: Enter Title (this can be left blank), Hit Enter

4\. Enter Creator (this can be left blank), Hit Enter

5\. Enter Mediatype (this can be left blank), Hit Enter

6\. Enter Delimiter, leave blank for ;, Hit Enter

5\. Hit Enter.

A csv file named with the search query will be in the same folder as the ia_search.py. 

The number of records should be printed out upon finishing. If the number of records is not equal to the number of rows (minus the header row) in the csv, there is a problem.

`150 records retrieved`

In [17]:
python ia_search.py

Keyword or Advanced (Title, Creator, Mediatype) Search (K/A): a
Enter Title: 
Enter Creator: Edgar Degas
Enter Mediatype (audio, collection, data, image, movies, texts, web): image
Enter Delimiter, leave blank for semi-colon: *
9 records retrieved
